### ISSUE TITLE
Verify Case Insensitive Emails

### ISSUE THREAD

**Issue by Magicjarvis** [gt-webdev/attendance#159]
Do so.

**Comment by thedekel** [comment]
emails also allow white-space in them... and don't require "@" and a period in them....

the database also has quite a few e-mails which are a single character long. Looks like I'm up for a night filled with hardcore database pruning... might as well make a web-interface for dealing with users and their silly data. Increasing the priority of this issue because it is too silly.

**Comment by thelastnode** [comment]
MongoLab has an interface already

**Comment by ash211** [comment]
Checking email validity with a regex based equivalent is ok for a quick
sanity check, but don't look for perfection on all the corner cases. Why
not just email a confirmation?

Sent from my mobile phone

On Sep 14, 2012, at 23:16, Ankit Shankar notifications@github.com wrote:

MongoLab has an interface already

On Sat, Sep 15, 2012 at 12:59 AM, Dekel notifications@github.com wrote:

> emails also allow white-space in them... and don't require "@" and a
> period in them....
> 
> the database also has quite a few e-mails which are a single character
> long. Looks like I'm up for a night filled with hardcore database
> pruning... might as well make a web-interface for dealing with users and
> their silly data. Increasing the priority of this issue because it is too
> silly.
> 
> —
> Reply to this email directly or view it on GitHub<
> https://github.com/gt-webdev/attendance/issues/159#issuecomment-8582172>.

—
Reply to this email directly or view it on
GitHubhttps://github.com/gt-webdev/attendance/issues/159#issuecomment-8582255.

**Comment by thedekel** [comment]
@thelastnode I was going to use that, but I need something that will also clear out any participations these users may have had in events. This would be a good point to also migrate the old system (arrays within the event objects) into the new system (separate "participation" objects). I've also been toying with the idea of extending "/admin" with something like a "/admin/users" that will allow admins to search through users (by regex,  events, orgs, etc.) and edit, delete, and merge (need to figure this one out) accounts.

@ash211 e-mail confirmations sound like a good idea (especially now that we upgraded our e-mail capacity), I'll open an issue for that.

And the reason for wanting better e-mail validation is to prevent common errors that we found recently. More specifically, I noticed that mobile keyboards will add a space after a "." if you have one in your e-mail address. I think that rather than just accepting this anyway and sending an e-mail that will never be read, we're better off returning the user to a pre-filled kiosk mode page and asking them to review it. Alternatively, we can just remove spaces.
